src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -2095,8 +2094,9 @@ private MemberDeclarationSyntax ParseMemberDeclarationOrStatementCore(SyntaxKind
                     var saveTerm = _termState;
                     _termState |= TerminatorState.IsPossibleStatementStartOrStop; // partial statements can abort if a new statement starts
 
-                    // Any expression is allowed, not just expression statements:
-                    var statement = this.ParseStatementNoDeclaration(allowAnyExpression: true);
+                    // We don't allow local declaration statements at the top level.  We want
+                    // to fall out below and parse them instead as fields.
+                    var statement = this.ParseStatementCore(isGlobalScriptLevel: true);
```

Comment:
the `allowAnyExpression` param name is not very helpful or intuitive as to what it is trying to convey.  really, it's only purpose is to say "we're at the global level, parse accordingly".